.github/workflows/build.yml

```diff
@@ -212,11 +216,56 @@ jobs:
       - name: Lint podspec
         run: bundle exec pod spec lint *.podspec --allow-warnings --verbose
 
+  check_version:
+    name: Check Version is Unreleased
+    runs-on: ubuntu-latest
+    outputs:
+      version: ${{ steps.version.outputs.version }}
+      tag: ${{ steps.tag_name.outputs.tag }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+      - name: Find Version
+        id: version
+        run: grep version gradle.properties >> $GITHUB_OUTPUT
+      - name: Generate Tag Name
+        id: tag_name
+        run: echo tag=v${{ steps.current_version.outputs.version }} >> $GITHUB_OUTPUT
+      - name: Check if version was already released
+        run: "! git ls-remote -t --exit-code origin ${{ steps.tag_name.outputs.tag }}"
+
+  tag_release:
+    name: Tag Release & Bump Version
+    runs-on: ubuntu-latest
+    if: github.event_name == 'workflow_dispatch' && inputs.triggerRelease && github.ref == 'refs/heads/main'
+    needs: [ build, check_metadata, ktlint, lint, android_tests, ios_tests, js_tests, check_version ]
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+      - name: Create Tag
+        uses: actions-ecosystem/action-push-tag@v1
+        with:
+          tag: ${{ needs.check_version.outputs.tag }}
+      - name: Bump Version
```

Comment:
I know this is the same name as before, but I wonder if there is anything we can do to make it more clear between this step and the `Update version for new development` step. From what I see, it seems that `Bump Version` is putting the version to what it needs to be for the release itself, and the update version step changes to the snapshot version for future development, is that right?

Reply:
I'll go re-evaluate the step names for that job

Reply:
I updated the step name to be "Generate next version"

Reply:
The part that is missing is that there is logic within the gradle build script that automatically appends `-SNAPSHOT` to the version unless the `releaseBuild` flag is set. So, the `gradle.properties` file already has the correct version in it for building the release.

This step just takes the version that was previously extracted from `gradle.properties` and increments it so that `gradle.properties` can be updated with it.